### PR TITLE
feat: [CDS-38810]: Azure Traffic Shift step

### PR DESCRIPTION
**Pull request by Tamaraius**
##### Summary:
Azure Traffic Shift step

##### Jira Links:
https://harness.atlassian.net/browse/CDS-38810

##### Screenshots:
https://user-images.githubusercontent.com/96610348/174073788-ee6d9366-e709-45bc-aba7-93a4b57f197a.mov

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by Tamaraius** [comment]
retrigger coverage

**Comment by Tamaraius** [comment]
retrigger coverage

**Comment by Tamaraius** [comment]
retrigger typecheck

**Comment by Tamaraius** [comment]
retrigger coverage

**Comment by Tamaraius** [comment]
retrigger typecheck

**Comment by Tamaraius** [comment]
retrigger build

**Comment by Tamaraius** [comment]
retrigger build

**Comment by Tamaraius** [comment]
trigger rebase

**Comment by Tamaraius** [comment]
retrigger build

**Comment by Tamaraius** [comment]
retrigger build

File: src/modules/75-cd/components/PipelineSteps/AzureTrafficShift/AzureTrafficShiftRef.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+import stepCss from '@pipeline/components/PipelineSteps/Steps/Steps.module.scss'
+
+export const AzureTrafficShiftRef = (props: AzureTrafficShiftProps, formikRef: StepFormikFowardRef): JSX.Element => {
+  /* istanbul ignore next */
+  const { allowableTypes, isNewStep = true, readonly, initialValues, onUpdate, onChange } = props
+  const { getString } = useStrings()
+  const { expressions } = useVariablesExpression()
+  const query = useQueryParams()
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const sectionId = (query as any).sectionId || ''
```
Comment (bhavya-sinha) [inline src/modules/75-cd/components/PipelineSteps/AzureTrafficShift/AzureTrafficShiftRef.tsx:33]: Can we have proper type for `query`

File: src/modules/75-cd/components/PipelineSteps/AzureTrafficShift/AzureTrafficShiftRef.tsx [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+      validationSchema={Yup.object().shape({
+        name: NameSchema({ requiredErrorMsg: getString('pipelineSteps.stepNameRequired') }),
+        timeout: getDurationValidationSchema({ minimum: '10s' }).required(getString('validation.timeout10SecMinimum')),
+        spec: Yup.object().shape({
+          traffic: Yup.string().required(getString('fieldRequired', { field: getString('pipeline.traffic') }))
+        })
+      })}
+    >
+      {formik => {
+        setFormikRef(formikRef, formik)
```
Comment (bhavya-sinha) [inline src/modules/75-cd/components/PipelineSteps/AzureTrafficShift/AzureTrafficShiftRef.tsx:63]: Why are we setting formil just on load?

**Review by bhavya-sinha — APPROVED** [review]

**Pull request merged**